src/app/shared/interceptors/cache.interceptor.ts

```diff
@@ -25,16 +25,16 @@ export class CacheInterceptor implements HttpInterceptor {
       return next.handle(req);
     }
 
-    const cachedResponse = this.cache.get(req);
-
     return Observable.create(observer => {
-      if (cachedResponse) {
-        return cachedResponse.subscribeOn(Scheduler.async).subscribe(observer);
+      let asyncResponse = this.cache.get(req);
```

Comment:
@christianvogt The only major change i can see here is the ` let asyncResponse = this.cache.get(req);` is inside `Observable.create`. Apart from that only the ordering of condition check is different. But i am not sure how this affects returning proper response?

Reply:
Aah, got it. Never mind. `Observable.create` used to close over the value of `cachedResponse` variable which is why it was not returning proper response.